conformance/utils/suite/suite.go

```diff
@@ -185,7 +187,11 @@ func NewConformanceTestSuite(options ConformanceOptions) (*ConformanceTestSuite,
 
 	roundTripper := options.RoundTripper
 	if roundTripper == nil {
-		roundTripper = &roundtripper.DefaultRoundTripper{Debug: options.Debug, TimeoutConfig: options.TimeoutConfig}
+		roundTripper = &roundtripper.DefaultRoundTripper{
+			Debug:             options.Debug,
+			TimeoutConfig:     options.TimeoutConfig,
+			CustomDialContext: options.RoundTripperDialContext,
```

Comment:
I'm not sure `CustomDialContext` belongs in the default without some kind of default value.  Removing it from the default would be fine.

Reply:
Hi @candita, thanks for the comment.  Please correct me if I'm wrong, but according to the `http.Transport` docs, a `nil` `DialContext` will result in the Dial method from the `net` package being used.


https://pkg.go.dev/net/http#Transport

```
	// DialContext specifies the dial function for creating unencrypted TCP connections.
	// If DialContext is nil (and the deprecated Dial below is also nil),
	// then the transport dials using package net.
	//
	// DialContext runs concurrently with calls to RoundTrip.
	// A RoundTrip call that initiates a dial may end up using
	// a connection dialed previously when the earlier connection
	// becomes idle before the later DialContext completes.
```